File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int buy = 0;
        int sell = 1;

        while (sell < prices.length) {
            if (prices[buy] < prices[sell]) {
                int prof = prices[sell] - prices[buy];
                max = Math.max(max, prof);
            } else {
                buy = sell;
            }
            sell++;
        }

        return max;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprof = 0;
        int b = 0;
        int s = 1;

        while (s < prices.size()) {
            if (prices[b] < prices[s]) {
                int prof = prices[s] - prices[b];
                maxprof = std::max(maxprof, prof);
            } else {
                b = s;
            }
            s++;
        }
        return maxprof;
    }
};
